sklearn/metrics/tests/test_classification.py

```diff
@@ -394,6 +397,79 @@ def test_confusion_matrix_multiclass_subset_labels():
                             [24, 3]])
 
 
+def test_multilabel_confusion_matrix():
+    """Test multilabel confusion matrix"""
+
+    # Non Regression test
+    y_true = [[0, 1, 1, 1, 1],
+              [1, 0, 0, 0, 0],
+              [0, 0, 1, 1, 1],
+              [1, 1, 1, 0, 0],
+              [0, 0, 0, 0, 1],
+              [1, 0, 1, 0, 0]]
+
+    y_pred = [[1, 1, 0, 0, 1],
+              [1, 0, 0, 1, 0],
+              [0, 0, 0, 1, 1],
+              [1, 1, 0, 0, 1],
+              [0, 0, 0, 1, 1],
+              [1, 1, 1, 0, 0]]
+
+    # Check against pre computed multilabel confusion matrix
+    #                   TP FP TN FN
+    result = np.array([[3, 1, 0, 2],   # <-- Label 0
+                       [2, 1, 0, 3],
+                       [1, 0, 3, 2],
+                       [1, 2, 1, 2],
+                       [3, 1, 0, 2]])  # <-- Label 4
+
+    assert_array_equal(result, multilabel_confusion_matrix(y_true, y_pred))
+
+    # Test sparse input
+    y_true = sparse.csr_matrix(y_true)
+    y_pred = sparse.coo_matrix(y_pred)
+
+    assert_array_equal(result, multilabel_confusion_matrix(y_true, y_pred))
+
+    # Sanity check for output shape
+    rng = np.random.RandomState(1)
+
+    n_samples, n_labels = np.random.randint(1, 200), np.random.randint(1, 100)
```

Comment:
I thought it would check for different sizes in a truly random fashion? If thats not desirable, could I just specify values instead?